ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -1906,6 +1923,7 @@ public void setCurrentCatalog(String currentCatalog) {
   }
 
   public void close() throws IOException {
+    sessionStates.remove(this);
```

Comment:
Please add new line between and maybe add a comment "Remove session state from the registry"